### PR TITLE
Jira Add Authentication

### PR DESCRIPTION
**Pull request by JasBeilin**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/42472

## Description
Added authentication using PAT

## Does it break backward compatibility?

   - [x] No

## Must have
- [x] Documentation

### CODE REVIEW

File: Packs/Jira/Integrations/JiraV2/JiraV2_description.md [hunk 1]
```diff
@@ -6,6 +6,13 @@ If you wish the first run to start from a specific time, use "Issue index to sta
...
+You can authenticate using one of the following methods:
+
+##### 1. Basic Authentication:
 As of June 2019, Basic authentication with passwords for Jira is no longer supported, please use basic authentication using API Token or use OAuth 1.0. To use basic authentication, follow [this tutorial](https://confluence.atlassian.com/cloud/api-tokens-938839638.html) to get the API token.
 
+##### 2. OAuth1.0:
 To use OAuth1.0 follow [this tutorial](https://developer.atlassian.com/cloud/jira/platform/jira-rest-api-oauth-authentication/) to get the Access Token. Authorizing using OAuth1.0, requires Access Token, Private Key, and Consumer Key. 
+
+##### 3. Private Access Token
+To use Private Access Token, follow [this tutorial](https://confluence.atlassian.com/enterprise/using-personal-access-tokens-1026032365.html) to get Access Tokens. Authorizing using PAT requires Access Token Only.
```
Comment (yuvalbenshalom) [inline Packs/Jira/Integrations/JiraV2/JiraV2_description.md:18]:
```suggestion
To use Private Access Token, follow [this tutorial](https://confluence.atlassian.com/enterprise/using-personal-access-tokens-1026032365.html) to get Access Tokens. Authorizing using PAT requires Access Token only.
Insert the generated token in the Access Token field.
```

File: Packs/Jira/Integrations/JiraV2/JiraV2.py [hunk 2]
```diff
@@ -112,10 +114,15 @@ def get_auth():
         HEADERS.update({'X-Atlassian-Token': 'nocheck'})
         return generate_oauth1()
 
+    elif is_bearer:
+        HEADERS.update({'Authorization': f'Bearer {access_token}'})
```
Comment (yuvalbenshalom) [inline Packs/Jira/Integrations/JiraV2/JiraV2.py:118]:
```suggestion
        #Personal Access Token Authentication
        HEADERS.update({'Authorization': f'Bearer {access_token}'})
```

**Review by yuvalbenshalom — CHANGES_REQUESTED** [review]
see my comments

**Review by yuvalbenshalom — APPROVED** [review]

File: Packs/Jira/Integrations/JiraV2/JiraV2.py [hunk 3]
```diff
@@ -112,10 +114,16 @@ def get_auth():
         HEADERS.update({'X-Atlassian-Token': 'nocheck'})
         return generate_oauth1()
 
+    elif is_bearer:
+        # Personal Access Token Authentication
```
Comment (anara123) [inline Packs/Jira/Integrations/JiraV2/JiraV2.py:118]: please add unit test

**Comment by JasBeilin** [comment]
@amshamah419 , can you please review the docs and make sure it's clear?

File: Packs/Jira/Integrations/JiraV2/JiraV2.yml [hunk 4]
```diff
@@ -20,11 +20,13 @@ configuration:
   required: false
   type: 4
 - defaultvalue: OauthKey
+  additionalinfo: If other required field are not provided (both Access Token and Private Key), will be ignored.
```
Comment (amshamah419) [inline Packs/Jira/Integrations/JiraV2/JiraV2.yml:23]:
```suggestion
  additionalinfo: Will be ignored if other required fields are not provided (for example both the Access Token and the Private Key).
```

File: Packs/Jira/Integrations/JiraV2/JiraV2_description.md [hunk 5]
```diff
@@ -6,6 +6,14 @@ If you wish the first run to start from a specific time, use "Issue index to sta
...
+You can authenticate using one of the following methods:
+
+##### 1. Basic Authentication:
 As of June 2019, Basic authentication with passwords for Jira is no longer supported, please use basic authentication using API Token or use OAuth 1.0. To use basic authentication, follow [this tutorial](https://confluence.atlassian.com/cloud/api-tokens-938839638.html) to get the API token.
 
+##### 2. OAuth1.0:
 To use OAuth1.0 follow [this tutorial](https://developer.atlassian.com/cloud/jira/platform/jira-rest-api-oauth-authentication/) to get the Access Token. Authorizing using OAuth1.0, requires Access Token, Private Key, and Consumer Key. 
+
+##### 3. Private Access Token
+To use Private Access Token, follow [this tutorial](https://confluence.atlassian.com/enterprise/using-personal-access-tokens-1026032365.html) to get Access Tokens. Authorizing using PAT requires Access Token only.
```
Comment (amshamah419) [inline Packs/Jira/Integrations/JiraV2/JiraV2_description.md:18]:
```suggestion
To use a Private Access Token, please follow [this tutorial](https://confluence.atlassian.com/enterprise/using-personal-access-tokens-1026032365.html) to obtain your Access Tokens. Authorizing using a PAT requires the Access Token only.
```

File: Packs/Jira/Integrations/JiraV2/JiraV2_description.md [hunk 6]
```diff
@@ -6,6 +6,14 @@ If you wish the first run to start from a specific time, use "Issue index to sta
...
+Insert the generated token in the Access Token field.
```
Comment (amshamah419) [inline Packs/Jira/Integrations/JiraV2/JiraV2_description.md:19]:
```suggestion
Insert the generated token in the `Access Token` field.
```

File: Packs/Jira/Integrations/JiraV2/README.md [hunk 7]
```diff
@@ -37,8 +37,8 @@ For more information about JQL syntax, go to https://www.atlassian.com/software/
     | Username (Basic Authentication) |  | False |
     | Password (Deprecated - Use API token) |  | False |
     | API token |  | False |
-    | Consumer key (OAuth 1.0) |  | False |
-    | Access token |  | False |
+    | Consumer key (OAuth 1.0) | If other required field are not provided (both Access Token and Private Key), will be ignored. | False |
```
Comment (amshamah419) [inline Packs/Jira/Integrations/JiraV2/README.md:40]:
```suggestion
    | Consumer key (OAuth 1.0) | Will be ignored if other required fields are not provided (for example both the Access Token and the Private Key). | False |
```

File: Packs/Jira/ReleaseNotes/1_3_9.md [hunk 8]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Atlassian Jira v2
+- Added *Personal Access Token* Authentication method.
```
Comment (amshamah419) [inline Packs/Jira/ReleaseNotes/1_3_9.md:4]:
```suggestion
- Added the *Personal Access Token* Authentication method.
```

**Review by amshamah419 — COMMENTED** [review]
See docs changes

**Pull request merged**